src/i2p.h

```diff
@@ -262,6 +275,12 @@ class Session
      * SAM session id.
      */
     std::string m_session_id GUARDED_BY(m_mutex);
+
+    /**
+     * Whether this is a transient session (the I2P private key will not be
+     * read or written to disk).
+     */
+    const bool m_transient;
```

Comment:
That guideline says "They are not useful" (`const` data members), but they improve readability - at a glance one can tell that this variable is only assigned in the constructor and never modified afterwards. Without the `const` one would have to manually check all references to the variable to find the places where it is modified.

Then the guideline says that `const` data members make the type difficult to use because it is not copy-assignable. But this is a feature and exactly what we want - to avoid copying a persistent session to a transient one, effectively converting it. Anyway, `i2p::sam::Session` is already non-copy-able, so adding a `const` data member does not change its copy-ability.